File: internal/content/testdata/bench/javascript/bubble-sort-swap-count.js
function bubble_sort_info(arr) {
    let a = arr.slice();
    let swaps = 0;
    let isSorted = false;
    while (!isSorted) {
        isSorted = true;
        for (let i = 0; i < a.length - 1; i++) {
            if (a[i] > a[i + 1]) {
                let tmp = a[i];
                a[i] = a[i + 1];
                a[i + 1] = tmp;
                swaps++;
                isSorted = false;
            }
        }
    }
    return {swaps: swaps, first: a[0], last: a[a.length - 1]};
}

File: internal/content/testdata/bench/java/bubble-sort-swap-count.java
import java.util.*;

class Solution {
    public Map<String, Object> bubble_sort_info(long[] arr) {
        long[] a = arr.clone();
        long swaps = 0;
        boolean isSorted = false;
        while (!isSorted) {
            isSorted = true;
            for (int i = 0; i < a.length - 1; i++) {
                if (a[i] > a[i + 1]) {
                    long t = a[i];
                    a[i] = a[i + 1];
                    a[i + 1] = t;
                    swaps++;
                    isSorted = false;
                }
            }
        }
        Map<String, Object> out = new LinkedHashMap<>();
        out.put("swaps", swaps);
        out.put("first", a[0]);
        out.put("last", a[a.length - 1]);
        return out;
    }
}
